### PR TITLE
Fix cli_tests websocket port binding issue #1178

### PR DESCRIPTION
**Pull request by abitmore**
PR for #1178.

Reference:
* https://unix.stackexchange.com/questions/305308/how-to-limit-range-of-random-port-sockets

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ script:
  - sed -i '/tests/d' libraries/fc/CMakeLists.txt
  - cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS=--coverage -DCMAKE_CXX_FLAGS=--coverage -DBoost_USE_STATIC_LIBS=OFF  -DCMAKE_CXX_OUTPUT_EXTENSION_REPLACE=ON .
  - 'which build-wrapper-linux-x86-64 && build-wrapper-linux-x86-64 --out-dir bw-output make -j 2 cli_wallet witness_node chain_test cli_test || make -j 2 cli_wallet witness_node chain_test cli_test'
+ - sudo sysctl -w net.ipv4.ip_local_port_range="32768 60999"
```
Comment (pmconrad) [inline .travis.yml]: Please remove that line, it's not necessary.

File: tests/cli/main.cpp [hunk 2]
```diff
@@ -71,7 +71,7 @@ int get_available_port()
    socklen_t len = sizeof(sin);
    if (getsockname(socket_fd, (struct sockaddr *)&sin, &len) == -1)
       return -1;
-   return sin.sin_port;
+   return ntohs(sin.sin_port);
```
Comment (pmconrad) [inline tests/cli/main.cpp]: While you're at it, please close socket_fd before returning.

**Comment by abitmore** [comment]
Rebased.

**Comment by pmconrad** [comment]
socket_fd remains open if either bind() or getsockname() fails, but I think in a unit test that's acceptable.

**Pull request merged**